AllReadyApp/Web-App/AllReady/Controllers/AccountController.cs

```diff
@@ -393,6 +400,7 @@ private void AddErrorsToModelState(IdentityResult result)
             }
         }
 
+        //TODO: either extract this into a provider pattern or somehow decide which of these conditionals are invoked by which caller and get rid of this method.

```

Comment:
@MisterJames, after further investigation, the `LocalRedirect` base method still does not solve the problem I'm trying to get at here.

For any action method that uses the existing `RedirectToLocal` method, this means that the unit test for that action method needs to test all four possible results of the outcome of the method. 

Since the method is called three times in the controller, and since there is no way of knowing whether or not some of the action methods in the controller invoke only some of the conditionals based on other criteria, it makes the AccountControllerTests class a mess.

A quick fix is to extract the `RedirectToLocal` method into its own class, encapsulating the conditional statements. When then write a test class class that tests the conditionals in the method.

This change frees AccountControllerTests from having to write four unit test methods for any action method that calls `RedirectToLocal` to account for the four possible execution paths.

A concrete example of this in AccountControllerTests is:
```
public async Task ExternalLoginCallbackRedirectsToCorrectActionAndController_WhenExternalLoginSignInAsyncResultIsSuccessful_AndExternalLoginInfoIsNotNull_AndUrlIsALocalUrl()
public async Task ExternalLoginCallbackRedirectsToCorrectActionAndController_WhenExternalLoginSignInAsyncResultIsSuccessful_AndExternalLoginInfoIsNotNull_AndUserIsASiteAdmin()
public async Task ExternalLoginCallbackRedirectsToCorrectActionAndControllerWithCorrerctRouteValues__WhenExternalLoginSignInAsyncResultIsSuccessful_AndExternalLoginInfoIsNotNull_AndUserIsAnOrgAdmin()
public async Task ExternalLoginCallbackRedirectsToCorrectActionAndController_WhenExternalLoginSignInAsyncResultIsSuccessful_AndExternalLoginInfoIsNotNull_AndUrlIsNotALocalUrl_AndUserTypeIsNotASiteAdminOrAnOrgAdmin()
```

Moving code from `RedirectToLocal` to a separate class would get rid of all these unneeded tests for all action method in AccountController.